File: 04.Simulation模拟/leetcode_3517_smallestPalindrome.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    string smallestPalindrome(string s) {
        int n = s.size();
        string left = s.substr(0, n/2);
        sort(left.begin(), left.end());
        left += (n%2 ? string(1, s[n/2]) : "");
        for (int i=n/2-1; i>=0; i--){
            left += left[i];
        }
        return left;
    }
};

int main(){
    Solution sl;
    string s = "baab";
    cout << sl.smallestPalindrome(s) << endl;
}

File: 04.Simulation模拟/leetcode_3517_smallestPalindrome.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        n = len(s)
        left = ''.join(sorted(s[:n//2]))
        return left+(s[n//2] if n%2 else "")+left[::-1]

if __name__ == "__main__":
    sl = Solution()
    s = "baab"
    print(sl.smallestPalindrome(s))
